Fix virt-v2v progress update

bkhizgiy opened the pull request:
After switching to use a Go wrapper, not all the information was redirected to the virt-v2v monitor command. Additionally, with changes done in the virt-v2v progress display, the progress was not reported to the controller. This patch fixes the piping to the virt-v2v monitoring command and the regex used to determine the copy disk progress of the migration.
https://github.com/kubev2v/forklift/pull/901

liranr23 commented on cmd/virt-v2v-monitor/virt-v2v-monitor.go:
```diff
@@ -11,11 +11,10 @@ import (
 
 var COPY_DISK_RE = regexp.MustCompile(`^.*Copying disk (\d+)/(\d+)`)
-var DISK_PROGRESS_RE = regexp.MustCompile(`^..\s*(\d+)% \[.*\]`)
+var DISK_PROGRESS_RE = regexp.MustCompile(`.+ (\d+)% \[[*-]+\]`)
```
are we sure we need to change the regex? i looked on this regex and the progress given in the log and it looked correct.

ahadas replied:
@liranr23 this change was inspired from https://github.com/oVirt/vdsm/pull/402
the problem with the current regex is that it catches lines that are unrelated to the progress

liranr23 reviewed (commented):
maybe some more info in the commit (and even the PR) should be nicer.

something like:
After refactoring the bash script into golang we missed using stderr as stdin to the v2v-monitor. Because of that, we didn't have the data progress reported.....

liranr23 commented:
please run `make bazel-generate`

ahadas commented on cmd/virt-v2v-monitor/virt-v2v-monitor.go:
```diff
@@ -95,37 +90,37 @@ func main() {
 		err := scanner.Err()
 		if err != nil {
-			klog.Fatal("Output monitoring failed! ", err)
+			fmt.Println("virt-v2v monitoring: Output monitoring failed! ", err)
```
klog.Fatal also calls OsExit which we lose with this change

bkhizgiy replied:
added os.Exit in addition.

ahadas commented on cmd/virt-v2v-monitor/virt-v2v-monitor.go:
```diff
@@ -95,37 +90,37 @@ func main() {
 			}
 		} else {
-			klog.V(1).Info("Ignoring line: ", string(line))
+			fmt.Println("virt-v2v monitoring: Ignoring line: ", string(line))
```
need to drop/change this - it creates too much noise

ahadas commented on cmd/virt-v2v-monitor/virt-v2v-monitor.go:
```diff
@@ -95,37 +90,37 @@ func main() {
 	err := scanner.Err()
 	if err != nil {
-		klog.Fatal("Output monitoring failed! ", err)
+		fmt.Println("virt-v2v monitoring: Output monitoring failed! ", err)
```
same comment as above for replacing klog.Fatal with a simple print

bkhizgiy replied:
same.

bkhizgiy commented:
@liranr23 ACK, I've updated the PR description and commit with more info.

liranr23 commented:
> After switching to use a Go wrapper, not all the information was redirected to the virt-v2v monitor command. Additionally, with changes done in the virt-v2v progress display, the progress was not reported to the controller. This patch fixes the piping to the virt-v2v monitoring command and the regex used to determine the copy disk progress of the migration. #901

great! but you changed it only on the PR and not on the commit?

bkhizgiy commented:
> > After switching to use a Go wrapper, not all the information was redirected to the virt-v2v monitor command. Additionally, with changes done in the virt-v2v progress display, the progress was not reported to the controller. This patch fixes the piping to the virt-v2v monitoring command and the regex used to determine the copy disk progress of the migration. #901
> 
> great! but you changed it only on the PR and not on the commit?

why, I've added `after switching to go wrapper` :) , I'm usually not adding to mush info into the commits, add some more details?

liranr23 commented:
> > > After switching to use a Go wrapper, not all the information was redirected to the virt-v2v monitor command. Additionally, with changes done in the virt-v2v progress display, the progress was not reported to the controller. This patch fixes the piping to the virt-v2v monitoring command and the regex used to determine the copy disk progress of the migration. #901
> > 
> > 
> > great! but you changed it only on the PR and not on the commit?
> 
> why, I've added `after switching to go wrapper` :) , I'm usually not adding to mush info into the commits, add some more details?

:)
That's OK. If you add it into the commit as a paragraph, it will allow others to look into git log and see it. When you don't do that, it will require to open the PR itself.

bkhizgiy commented:
> > > > After switching to use a Go wrapper, not all the information was redirected to the virt-v2v monitor command. Additionally, with changes done in the virt-v2v progress display, the progress was not reported to the controller. This patch fixes the piping to the virt-v2v monitoring command and the regex used to determine the copy disk progress of the migration. #901
> > > 
> > > 
> > > great! but you changed it only on the PR and not on the commit?
> > 
> > 
> > why, I've added `after switching to go wrapper` :) , I'm usually not adding to mush info into the commits, add some more details?
> 
> :) That's OK. If you add it into the commit as a paragraph, it will allow others to look into git log and see it. When you don't do that, it will require to open the PR itself.

Added to the commit as well:)

ahadas commented on virt-v2v/cold/entrypoint.go:
```diff
@@ -213,9 +208,8 @@ func executeVirtV2v(args []string) (err error) {
-	virtV2vMonitorCmd.Stdin = tee
+	virtV2vMonitorCmd.Stdin = r
 	virtV2vMonitorCmd.Stdout = os.Stdout
-	virtV2vMonitorCmd.Stderr = os.Stderr
```
was it done intentionally?
